File: Desktop/23-06-2020-jeyhunkhh-master/Practice/Car.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Practice
{
    class Car
    {
        public double CurrentFuel { get; private set; }
        public double FuelUsage { get; private set; }
        public double FuelCapacity { get; private set; }
        public double LocalKm { get; private set; }
        public double GlobalKm { get; private set; }

        public Car(double fuelUsage,double fuelCapacity)
        {
            this.FuelUsage = fuelUsage;
            this.FuelCapacity = fuelCapacity;

            this.CurrentFuel = this.FuelCapacity * 0.05;
        }

        public bool Drive(double km)
        {
            double usageLitr = this.FuelUsage * km / 100;

            if (usageLitr > this.CurrentFuel) return false;

            this.CurrentFuel -= usageLitr;
            this.GlobalKm += km;
            this.LocalKm += km;
            return true;
        }

        public bool AddFuel(double litr)
        {
            if (this.CurrentFuel + litr > this.FuelCapacity) return false;

            this.CurrentFuel += litr;
            return true;
        }

        public void ResetLocalKm()
        {
            this.LocalKm = 0;
        }
    }
}

File: Desktop/DERS/Masin, Silah/js/car.js
class Car{
    constructor(fuelUsage,fuelCapacity){
        this.fuelUsage = fuelUsage;
        this.fuelCapacity = fuelCapacity;
        this.currentFuel = 0;
        this.localKm = 0;
        this.globalKm = 0;
    }

    drive(km){
        if (km * this.fuelUsage / 100 > this.currentFuel) {
            console.log("Cari yanacaqla bu qədər yolu gedə bilməzsiniz");
            return;
        }

        this.currentFuel -= km * this.fuelUsage / 100;
        this.localKm += km;
        this.globalKm += km;

        console.log(`${km} km maşın sürüldü, cari yanacaq : ${this.currentFuel} litrdir`);
    }

    addFuel(litr){
        if (litr + this.currentFuel > this.fuelCapacity) {
            console.log("Bu qədər yanacaq əlavə edə bilməzsiniz")
            return;
        }

        this.currentFuel += litr;
        console.log(`${litr} litr yanacaq əlavə olundu, cari yanacaq : ${this.currentFuel} litrdir`);
    }

    resetLocalKm() {
        this.localKm = 0;
        console.log("Maşının localkm-i sıfırlandı");
    }
}

let c180 = new Car(11,45);
c180.addFuel(20);
c180.drive(20)
c180.addFuel(30);

console.log(c180);

let a4 = new Car(13,50);
a4.addFuel(30);
a4.drive(50);
a4.resetLocalKm();

console.log(a4);
